services/github/github-total-star.service.js

```diff
@@ -0,0 +1,259 @@
+'use strict'
+
+const Joi = require('@hapi/joi')
+const gql = require('graphql-tag')
+const { nonNegativeInteger } = require('../validators')
+const { metric } = require('../text-formatters')
+const { GithubAuthV4Service } = require('./github-auth-service')
+const {
+  documentation: commonDocumentation,
+  transformGraphqlErrors: transformErrors,
+} = require('./github-helpers')
+
+const MAX_REPO_LIMIT = 200
+
+const customDocumentation = `This badge takes upto <mark>${MAX_REPO_LIMIT}</mark> most starred repositories into account.`
+
+const userDocumentation = `${commonDocumentation}
+<p>
+  <b>Note:</b><br>
+  1. ${customDocumentation}<br>
+  2. <mark>affiliations</mark> query param accepts three value <mark>OWNER</mark>, <mark>COLLABORATOR</mark>, <mark>ORGANIZATION_MEMBER</mark>.
+  One can pass comma separated combination of these values (no spaces) e.g. <mark>OWNER,COLLABORATOR</mark> or <mark>OWNER,COLLABORATOR,ORGANIZATION_MEMBER</mark>.
+  Default value is <mark>OWNER</mark>. See these values explanation <a href="https://docs.github.com/en/graphql/reference/enums#repositoryaffiliation">here</a>.
+</p>
+`
+const orgDocumentation = `${commonDocumentation}
+<p>
+  <b>Note:</b> ${customDocumentation}
+</p>`
+
+const pageInfoSchema = Joi.object({
+  hasNextPage: Joi.boolean().required(),
+  endCursor: Joi.string().allow(null).required(),
+}).required()
+
+const nodesSchema = Joi.array()
+  .items(
+    Joi.object({
+      stargazers: Joi.object({
+        totalCount: nonNegativeInteger.required(),
```

Comment:
No need to add the `.required()` here, as it's already included by `nonNegativeInteger` (`optionalNonNegativeInteger` is the optional version of this)
```suggestion
        totalCount: nonNegativeInteger,
```

Reply:
Done